react added into the project

Willian1661 opened the pull request:
- added the react packages
- set up the webpack configurations to run react
i take so long to finish this because i was getting an error with the webpack config and with the css's plugins

close #24

eerison commented on src/components/buttons-group.js:
```diff
@@ -1,7 +1,7 @@
+function buttonsGroup(groupsId,buttons,htmlClass) {		
 	return `
-		<div id="${divsId}" class="${htmlClass}">
+		<section id="${groupsId}" class="${htmlClass}">
```
It's not needed

eerison replied:
no make any sense use section inside of button group, it's not a `section`

my last comment was just an example, in case you change div to other that like `p`, `span`, and so on!

in this case doesn't make sense you receive `divsId`, because it could change to other thing that isn't a div!

eerison commented on src/components/calculator.js:
```diff
@@ -15,54 +15,56 @@ const plusMinus = require("./icons/plus-minus-icon");
-		button(plusMinus(), "button is-large is-light m-2 is-fullwidth is-one-fifth has-text-warning", "+/-"),
-		button(delet(), "button is-large is-light m-2 is-fullwidth is-one-fifth has-text-warning", "D"), 
-		button(percent(), "button is-large is-light m-2 is-fullwidth is-one-fifth has-text-warning", "%")
+		button(clearAll(), "is-large is-light m-2 is-fullwidth is-one-fifth has-text-danger is-mobile", "C"), 
```
Please just do changes related with the ticket

Willian1661 replied:
i know, but when you merged the last pr i did a few changes that i forgot to commit and because was pending things i did inside the same ticket, how can i do this without do it inside a non respective ticket?

eerison replied:
open a new PR for those changes!

Willian1661 replied:
ok i get it

Willian1661 replied:
ok i get it

eerison reviewed (changes requested):
Please remove all changes that are not related with react

Willian1661 commented:
> Please remove all changes that are not related with react

ok

eerison commented:
put the ticket as draft, just put it as `open` when the work is **done**

Willian1661 committed:
react added into the project

Willian1661 committed:
- recreated the components with React
- recreated tests of the changed components
- added react-devTools package
- changed main code

eerison commented on package.json:
```diff
@@ -52,6 +52,8 @@
     "lodash": "^4.17.21",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-devtools": "^4.27.1",
```
it should be inside of dev dependencies

Willian1661 replied:
done

eerison commented on src/components/Button.js:
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+export default function Button(props) {
```
hmm usually I see components like this

```js
export default function Button({htmlClass, data, children}) {
```

Maybe it's for you don't type `props` every time

Willian1661 replied:
yea could be,is better don't use props with single elements? likee button or something like this

Willian1661 replied:
done

eerison commented on src/components/Calculator.js:
```diff
@@ -0,0 +1,103 @@
+							groupsId="buttonsFirstLine" htmlClass="columns is-justify-content-center">
+
+							<Button htmlClass="is-large is-light m-2 is-fullwidth is-one-fifth has-text-danger" data="C">
+								<i class='icon fa-solid fa-c'></i>
```
Where are the icons components?

Willian1661 replied:
i did'nt create yeat, because i wanted to import them from a package
that's the reason i want to create another pr, to don't this pr that is for react with the icons stuff

Willian1661 replied:
i think i'm gonna create the icons component normally, after merge this pr i'm gonna past them to react and install the packages

Willian1661 replied:
done

Willian1661 committed:
changed icons components to React

eerison commented on src/components/icons/clear-all-icon.js:
```diff
@@ -1,4 +1,6 @@
-	return "<i class='icon fa-solid fa-c'></i>";
+import React from "react";
+
+export default function iconClearAll() {
```
Follow the same way that you made for Button component

```js
<IconClearAll />
```

eerison replied:
Ok, But in button component you start with Capital letter

Willian1661 replied:
what you mean? `<iconClear />` i'm not importing any package to return a tag

Willian1661 replied:
hmmm a tested here, i didn't know this would work 🤔

eerison replied:
I know, I mean that you function name should be `IconClearAll`

Willian1661 replied:
🤦‍♂️ i changed the name but deleted the change by accident and i didn't even noticed, next commit i change this again
it's right now

Willian1661 replied:
done

eerison commented on src/components/icons/clear-all-icon.js:
```diff
@@ -1,4 +1,6 @@
-	return "<i class='icon fa-solid fa-c'></i>";
+import React from "react";
+
+export default function iconClearAll() {
```
Use components files like

`AsteriskIcon.js`


change this for all components

Willian1661 replied:
but using Icon {name} it's a pattern of react-icons

eerison replied:
wait here I'm talking about file name

Willian1661 replied:
aah ok

Willian1661 replied:
done

eerison commented on src/components/Buttons-group.js:
```diff
@@ -0,0 +1,9 @@
+import React from "react";
```
Rename to `ButtonsGroup.js`

Willian1661 replied:
done

Willian1661 committed:
renamed all the components and tests

Willian1661 commented:
@eerison 
I was testing the react icons and it seems that we still can use the **font awesome** icons by react icons :p

Willian1661 committed:
renamed icons component
